independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/RequestContext.java

```diff
@@ -172,7 +172,7 @@ public ContextState getState() {
     }
 
     public ContextState getStateIfActive() {
-        return currentContext.get();
+        return isActive() ? currentContext.get() : null;
```

Comment:
I think that it would be better to do something like:
```java
ContextState state = currentContext.get();
return state != null ? state.isValid() : false;
```

Reply:
oki, np

Reply:
done